pkg/cloud/google/machineactuator.go

```diff
@@ -796,19 +800,51 @@ func getOrNewComputeServiceForMachine(params MachineActuatorParams) (GCEClientCo
 	if params.ComputeService != nil {
 		return params.ComputeService, nil
 	}
-	// The default GCP client expects the environment variable
-	// GOOGLE_APPLICATION_CREDENTIALS to point to a file with service credentials.
-	client, err := google.DefaultClient(context.TODO(), compute.ComputeScope)
-	if err != nil {
-		return nil, err
+
+	var client *http.Client
+	var err error
+	// If specified in the GCE config, use the alternative authentication.
+	if params.CloudConfigPath != "" {
+		glog.Info("Trying to get open the GCE config")
+		client, err = clientWithAltTokenSource(params.CloudConfigPath)
 	}
+	if params.CloudConfigPath == "" || err != nil {
+		glog.Info("Using the default GCP client")
+		if err != nil {
```

Comment:
The nesting of this check seems weird, since the above log line will print before this warning.

Reply:
Disappeared because of the above comment.